### PR TITLE
146 ci cd on every pull request, relief heavy burdern

### PR DESCRIPTION
**Pull request by thienguen**
# What type of PR is this? (Check all that apply)

- [ ] ✨**feature**: Introduces completely new code or new features.
- [ ] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [ ] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [x] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [x] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

What did you change? How did you change it?

# Tests

How was this tested?

- [ ] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

Explanation:

You should be able to see beside the approval of this pull request, there is another run for code coverage statistic of us

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [ ] NO documentation needed

Link to external documentation:

# [Optional] Screenshots

image

image

image

# [Optional] Are there any post-deployment tasks we need to perform?

Yes, you might have to turn on the second workflow, FOR LINTING, FORMATTING, and BEST PRACTICES that can be caught through code scanning, we have had this for a while, but since we have members still indent in 4 spaces, I put this on hold till we push the last 2 weeks

image

- [ ] 1. navigate to actions tab
- [ ] 2. choose the 1st one
- [ ] 3. manually trigger it
- [ ] 4. choose the right branch

image

image

Make sure you choose the branch you are on, this is for every pull request being made

# Additional notes

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Barbel Github Action
+name: Barbel Best Practices
```
Comment (taylorfinelli) [inline .github/workflows/ci.yml:1]: Typo: Barbel -> Barbell

**Review by taylorfinelli — DISMISSED** [review]

**Comment by taylorfinelli** [comment]
Fix typo then I can approve

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mva919) [inline .github/workflows/ci.yml:1]: Lets rename this to `Barbell Code Checker` since this workflow does code checking and formatting.

**Review by mva919 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (thienguen) [reply .github/workflows/ci.yml]: fixed

Comment (thienguen) [reply .github/workflows/ci.yml]: fixed

**Review by mva919 — APPROVED** [review]
Thanks for making the changes

**Pull request merged**